packages/mobile/src/fiatExchanges/FiatExchangeOptions.tsx

```diff
@@ -59,14 +78,55 @@ export const fiatExchangesOptionsScreenOptions = ({
 }
 
 function FiatExchangeOptions({ route, navigation }: Props) {
+  const [providerLocalAmount, setProviderLocalAmount] = React.useState<BigNumber>()
+  const [providerlocalCurrency, setProviderLocalCurrency] = React.useState<LocalCurrencyCode>()
+  // All currency exchange calculations handled here as
+  // different providers may provide different fiat, CELO and cUSD rates
+
+  const localCurrency = useSelector(getLocalCurrencyCode)
   const { amount } = route.params
+
   const goToProvider = (screen: keyof StackParamList) => {
-    return () => navigation.navigate(screen, { amount })
+    return () =>
+      navigation.navigate(screen, {
+        localAmount: features.CUSD_MOONPAY_ENABLED
+          ? amount
+          : providerLocalAmount || new BigNumber(0),
+        currencyCode: providerlocalCurrency || FALLBACK_CURRENCY,
+      })
   }
   const onDismiss = () => {
     navigation.setParams({ isExplanationOpen: false })
   }
 
+  const dispatch = useDispatch()
+  React.useEffect(() => {
+    async function getMoonpayRates() {
```

Comment:
I wonder if this structure here could be simplified by using one of the hook utils we already have for doing async requests. For example see Jean's use of `useAsync` here: https://github.com/celo-org/celo-monorepo/blob/master/packages/mobile/src/fees/CalculateFee.tsx#L68

Reply:
Oh sweet, hadn't seen that library. Simplified the function a bit but will wait to use the `react-async-hook` until we start using the local rates in the display